grype/presenter/explain/explain.go

```diff
@@ -0,0 +1,467 @@
+package explain
+
+import (
+	_ "embed"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/anchore/grype/grype/match"
+	"github.com/anchore/grype/grype/pkg"
+	"github.com/anchore/grype/grype/presenter/models"
+	"github.com/anchore/syft/syft/file"
+)
+
+//go:embed explain_cve.tmpl
+var explainTemplate string
+
+type VulnerabilityExplainer interface {
+	ExplainByID(IDs []string) error
+	ExplainBySeverity(severity string) error
+	ExplainAll() error
+}
+
+type ViewModel struct {
+	PrimaryVulnerability   models.VulnerabilityMetadata
+	RelatedVulnerabilities []models.VulnerabilityMetadata
+	MatchedPackages        []*explainedPackage // I think this needs a map of artifacts to explained evidence
+	URLs                   []string
+}
+
+type viewModelBuilder struct {
+	// TODO: this field is unused. Need think a bit more here.
```

Comment:
Do we want to remove this field if it's unused now? Are there still design aspects being thought through for exposing this in a different way?